stable_baselines3/common/base_class.py

```diff
@@ -628,6 +628,8 @@ def set_parameters(
                 attr.load_state_dict(params[name])  # type: ignore[arg-type]
             else:
                 # Assume attr is th.nn.Module
+                # Truncate parameter name prefixes if state_dict comes from compiled model
+                params = utils.rename_torch_compile_parameters(params, name)
```

Comment:
we should probably only do that if the policy is compiled, `isinstance()` check should be helpful there.

Reply:
This is not in scope of how set_parameters is used, here. First, the policy is not existing, when loading from the class method, second, we do not have access in the function and the algorithm's policy attribute may not be compiled when loading parameter values. Since you replied to a prior PR that policies should be compiled externally on user request, introducing a compiled flag on loading or algorithm/policy attribute is infeasible.

Reply:
sorry, here I meant we should do the renaming when saving the policy if possible (because at that time, we should know about whether it is compiled or not).

And as I wrote in https://github.com/DLR-RM/stable-baselines3/issues/1438#issuecomment-1516132776, I 'm surprised that pytorch doesn't have anything to handle that, it sounds like a common issue.

Reply:
> we should do the renaming when saving the policy if possible (

if not possible, then we can keep the current code